### PR TITLE
Fix authentication issues in Az.Accounts 2.1

### PR DESCRIPTION
**Pull request by erich-wang**
## Description

* Fixed the issue that token is not renewed after expiring for LRO [#13367]
* Fixed the issue that AccountId is not respected in MSI [#13376]
* Fixed the issue that error message is unclear if browser is not avaialable for Interactive auth [#13340]
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Accounts/Authenticators/MsalAccessToken.cs [hunk 1]
```diff
@@ -39,50 +39,120 @@ public class MsalAccessToken : IAccessToken
...
-        public MsalAccessToken(string token, string tenantId, string userId = null, string homeAccountId = null)
+        private DateTimeOffset ExpiredOn { get; set; }
+
+        private readonly static TimeSpan ExpirationThreshold = TimeSpan.FromMinutes(5);
+
+        private TokenCredential TokenCredential { get; set; }
+
+        private TokenRequestContext TokenRequestContext { get; set; }
+
+        //public MsalAccessToken(string token, DateTimeOffset expiresOn, string tenantId, string userId = null, string homeAccountId = null)
```
Comment (dingmeng-xue) [inline src/Accounts/Authenticators/MsalAccessToken.cs:50]: Why not remove this code?

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs [hunk 2]
```diff
@@ -398,12 +401,35 @@ public override void ExecuteCmdlet()
...
+                           throw;
+                       }
+                   }
                });
             }
         }
 
+        private bool IsUableToOpenWebPageError(AuthenticationFailedException exception)
+        {
+            return exception.InnerException is MsalClientException && ((MsalClientException)exception.InnerException)?.ErrorCode == "linux_xdg_open_failed"
```
Comment (dingmeng-xue) [inline src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:429]: Is there const defined in MSAL?

**Review by dingmeng-xue — DISMISSED** [review]

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs [hunk 3]
```diff
@@ -398,12 +401,35 @@ public override void ExecuteCmdlet()
...
+                       }
+                   }
                });
             }
         }
 
+        private bool IsUableToOpenWebPageError(AuthenticationFailedException exception)
+        {
+            return exception.InnerException is MsalClientException && ((MsalClientException)exception.InnerException)?.ErrorCode == MsalError.LinuxXdgOpen
+                            || (exception.Message?.ToLower()?.Contains("uable to open a web page") ?? false);
```
Comment (isra-fel) [inline src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:430]: A typo. Original message: "Unable to open a web page using xdg-open."

```suggestion
                            || (exception.Message?.ToLower()?.Contains("unable to open a web page") ?? false);
```

File: src/Accounts/Authenticators/MsalAccessToken.cs [hunk 4]
```diff
@@ -39,50 +39,79 @@ public class MsalAccessToken : IAccessToken
...
+        {
+            if(IsExpired())
+            {
+                var token = TokenCredential.GetToken(TokenRequestContext, default(CancellationToken));
+                AccessToken = token.Token;
+                ExpiredOn = token.ExpiresOn;
+            }
+        }
+
+        private bool IsExpired()
```
Comment (isra-fel) [inline src/Accounts/Authenticators/MsalAccessToken.cs:105]: Suggest rename to match the inner logic
```suggestion
        private bool IsNearExpiration()
```

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs (hunk 3 above)
Comment (erich-wang) [reply src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:430]: Good catch, I just noticed as well, thx.

**Review by isra-fel — APPROVED** [review]
LGTM.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**